test/celeritas/ext/G4VecgeomConversion.test.cc

```diff
@@ -0,0 +1,183 @@
+    EXPECT_EQ("Envelope", geom.id_to_label(VolumeId{1}));
+    EXPECT_EQ("Shape1", geom.id_to_label(VolumeId{2}));
+    EXPECT_EQ("Shape2", geom.id_to_label(VolumeId{3}));
+}
+
+//---------------------------------------------------------------------------//
+
+TEST_F(G4VecgeomConvTest, host_access_vecgeom)
+{
+    // access world volumes through GeoParams object
+    const auto& geom = *this->geom_.get();
+
+    // const auto& world = geom.host_ref();
+    //  get uniform pointer to PlacedVolume
+    // PlacedVolumeT<MemSpace::host> const& world{*pworld};
+    const auto& geoparams = geom.host_ref();
+    EXPECT_TRUE(geoparams);
+
+    // address to VecGeom objects
+    const auto& world = *geoparams.world_volume;
+    EXPECT_TRUE(world.GetLogicalVolume());
+    EXPECT_EQ(8, world.GetDaughters().size());
+
+    // world->PrintContent();
+}
+
+//---------------------------------------------------------------------------//
+
+#include "corecel/data/CollectionStateStore.hh"
+
+#include "G4VecgeomConversion.test.hh" // for definition of VGGTestInput,Output structs
+
+// Since VecGeom is currently CUDA-only, we cannot use the TEST_IF_CELER_DEVICE
+// macro (which also allows HIP).
+#if CELERITAS_USE_CUDA
+#    define TEST_IF_CELERITAS_CUDA(name) name
+#else
+#    define TEST_IF_CELERITAS_CUDA(name) DISABLED_##name
+#endif
+
+TEST_F(G4VecgeomConvTest, TEST_IF_CELERITAS_CUDA(device_tracking))
+{
+    using StateStore = CollectionStateStore<VecgeomStateData, MemSpace::device>;
+
+    // Set up test input
+    // TODO: apply mm -> cm conversion to volume dimensions in the G4->VecGeom
+    // conversion
+    celeritas_test::G4VGConvTestInput input;
+    input.init = {{{100, 100, 100}, {1, 0, 0}},
+                  {{100, 100, -100}, {1, 0, 0}},
+                  {{100, -100, 100}, {1, 0, 0}},
+                  {{100, -100, -100}, {1, 0, 0}},
+                  {{-100, 100, 100}, {-1, 0, 0}},
+                  {{-100, 100, -100}, {-1, 0, 0}},
+                  {{-100, -100, 100}, {-1, 0, 0}},
+                  {{-100, -100, -100}, {-1, 0, 0}}};
+
+    const auto& geoparams = *this->geom_.get();
+    StateStore  device_states(geoparams.host_ref(), input.init.size());
+    input.max_segments = 4;
+    input.params       = geoparams.device_ref();
+    input.state        = device_states.ref();
+
+    // Run kernel
+    auto output = g4vgconv_test(input);
+
+    static const int expected_ids[] = {2,  1,  0, -1, 2,  1,  0, -1, 2,  1, 0,
+                                       -1, 2,  1, 0,  -1, 2,  1, 0,  -1, 2, 1,
+                                       0,  -1, 2, 1,  0,  -1, 2, 1,  0,  -1};
+
+    static const double expected_distances[]
+        = {50, 10, 10, 70, 50, 10, 10, 70, 50, 10, 10, 70, 50, 10, 10, 70,
+           50, 10, 10, 70, 50, 10, 10, 70, 50, 10, 10, 70, 50, 10, 10, 70};
+
+    // Check results
+    EXPECT_VEC_EQ(expected_ids, output.ids);
+    EXPECT_VEC_SOFT_EQ(expected_distances, output.distances);
+}
```

Comment:
Since none of these tests directly test `G4VecgeomConverter` (they just test the constructed geometry), and I think the CUDA tracing code is identical to the `VecGeomParams.test.cu` code, I think these tests should be moved to the `VecGeomParams.test.cc` file, and in the `CMakeLists.txt` you can use the `FILTER` keyword to `celeritas_add_device_test` for running separately the two test fixtures.

Reply:
I think it is important to leave one such tracking test as verification of the consistency of the geometry conversion.

Reply:
Yes, I agree, but they should be included as part of the Vecgeom test to reduce code duplication and because that's the only class directly being tested.